### ISSUE TITLE
Right click on editor tabs and selecting New window launches window then dies.

### ISSUE THREAD

**Issue by Sa-Bender17** [spyder-ide/spyder#20275]
### What steps will reproduce the problem?

Open spyder from Windows cmnd line.
Right click and select New Window from context popup for
tabs of editor view.
Window pops up briefly, then this error report window comes up.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\swaredvp\py_envtry\SPYDER\lib\site-packages\spyder\plugins\outlineexplorer\widgets.py", line 550, in update_editor
    if not self.starting[language]:
KeyError: 'python'
```

## Versions

* Spyder version: 5.4.0  (pip)
* Python version: 3.10.8 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.1 (OK)
chardet >=2.0.0               :  5.1.0 (OK)
cloudpickle >=0.5.0           :  2.2.0 (OK)
cookiecutter >=1.6.0          :  2.1.1 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.34.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.2 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.17.3 (OK)
keyring >=17.0.0              :  23.13.1 (OK)
nbconvert >=4.0               :  7.2.7 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
paramiko >=2.4.0              :  2.12.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.4 (OK)
pygments >=2.0                :  2.13.0 (OK)
pylint >=2.5.0;<3.0           :  2.15.9 (OK)
pylint_venv >=2.1.1           :  2.3.0 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.6.0;<1.7.0          :  1.6.0 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.2.2             :  0.2.2 (OK)
qtawesome >=1.2.1             :  1.2.2 (OK)
qtconsole >=5.4.0;<5.5.0      :  5.4.0 (OK)
qtpy >=2.1.0                  :  2.3.0 (OK)
rtree >=0.9.7                 :  1.0.1 (OK)
setuptools >=49.6.0           :  63.2.0 (OK)
sphinx >=0.6.6                :  5.3.0 (OK)
spyder_kernels >=2.4.0;<2.5.0 :  2.4.1 (OK)
textdistance >=4.2.0          :  4.5.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.2.0 (OK)
zmq >=22.1.0                  :  24.0.1 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  None (NOK)
numpy >=1.7                   :  None (NOK)
pandas >=1.1.1                :  None (NOK)
scipy >=0.17.0                :  None (NOK)
sympy >=0.7.3                 :  None (NOK)
```

### Environment

<details><summary>Environment</summary>

```
Package                       Version
----------------------------- -----------
alabaster                     0.7.12
arrow                         1.2.3
astroid                       2.12.13
atomicwrites                  1.4.1
attrs                         22.2.0
autopep8                      1.6.0
Babel                         2.11.0
backcall                      0.2.0
bcrypt                        4.0.1
beautifulsoup4                4.11.1
binaryornot                   0.4.4
black                         22.12.0
bleach                        5.0.1
certifi                       2022.12.7
cffi                          1.15.1
chardet                       5.1.0
charset-normalizer            2.1.1
click                         8.1.3
cloudpickle                   2.2.0
colorama                      0.4.6
comm                          0.1.2
cookiecutter                  2.1.1
cryptography                  38.0.4
debugpy                       1.6.4
decorator                     5.1.1
defusedxml                    0.7.1
diff-match-patch              20200713
dill                          0.3.6
docstring-to-markdown         0.11
docutils                      0.19
entrypoints                   0.4
fastjsonschema                2.16.2
flake8                        5.0.4
idna                          3.4
imagesize                     1.4.1
importlib-metadata            5.2.0
inflection                    0.5.1
intervaltree                  3.1.0
ipykernel                     6.19.4
ipython                       7.34.0
ipython-genutils              0.2.0
isort                         5.11.4
jaraco.classes                3.2.3
jedi                          0.18.2
jellyfish                     0.9.0
Jinja2                        3.1.2
jinja2-time                   0.2.0
jsonschema                    4.17.3
jupyter_client                7.4.8
jupyter_core                  5.1.1
jupyterlab-pygments           0.2.2
keyring                       23.13.1
lazy-object-proxy             1.8.0
MarkupSafe                    2.1.1
matplotlib-inline             0.1.6
mccabe                        0.7.0
mistune                       2.0.4
more-itertools                9.0.0
mypy-extensions               0.4.3
nbclient                      0.7.2
nbconvert                     7.2.7
nbformat                      5.7.1
nest-asyncio                  1.5.6
numpydoc                      1.5.0
packaging                     22.0
pandocfilters                 1.5.0
paramiko                      2.12.0
parso                         0.8.3
pathspec                      0.10.3
pexpect                       4.8.0
pickleshare                   0.7.5
pip                           22.3.1
platformdirs                  2.6.2
pluggy                        1.0.0
prompt-toolkit                3.0.36
psutil                        5.9.4
ptyprocess                    0.7.0
pycodestyle                   2.9.1
pycparser                     2.21
pydocstyle                    6.1.1
pyflakes                      2.5.0
Pygments                      2.13.0
pylint                        2.15.9
pylint-venv                   2.3.0
pyls-spyder                   0.4.0
PyNaCl                        1.5.0
PyQt5                         5.15.7
PyQt5-Qt5                     5.15.2
PyQt5-sip                     12.11.0
PyQtWebEngine                 5.15.6
PyQtWebEngine-Qt5             5.15.2
pyrsistent                    0.19.3
python-dateutil               2.8.2
python-lsp-black              1.2.1
python-lsp-jsonrpc            1.0.0
python-lsp-server             1.6.0
python-slugify                7.0.0
pytoolconfig                  1.2.4
pytz                          2022.7
pywin32                       305
pywin32-ctypes                0.2.0
PyYAML                        6.0
pyzmq                         24.0.1
QDarkStyle                    3.0.3
qstylizer                     0.2.2
QtAwesome                     1.2.2
qtconsole                     5.4.0
QtPy                          2.3.0
requests                      2.28.1
rope                          1.6.0
Rtree                         1.0.1
setuptools                    63.2.0
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.3.2.post1
Sphinx                        5.3.0
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
spyder                        5.4.0
spyder-kernels                2.4.1
text-unidecode                1.3
textdistance                  4.5.0
three-merge                   0.1.1
tinycss2                      1.2.1
toml                          0.10.2
tomli                         2.0.1
tomlkit                       0.11.6
tornado                       6.2
traitlets                     5.8.0
ujson                         5.6.0
urllib3                       1.26.13
watchdog                      2.2.0
wcwidth                       0.2.5
webencodings                  0.5.1
whatthepatch                  1.0.3
wrapt                         1.14.1
yapf                          0.32.0
zipp                          3.11.0

```
</details>

**Comment by ccordoba12** [comment]
Hey @Sa-Bender17, thanks for reporting. This is a duplicate of issue #20101 and it'll be fixed in our next version (5.4.1), to be released later today.

**Comment by Sa-Bender17** [comment]
Thanks for the quick response. I love the tool, it’s quickly becoming my Python IDE of choice.



Sincerely,

Steven Bender

Software Engineer
ICM Controls
(315) 233-5266 ext 313
(315) 362-9313 direct
***@***.******@***.***>
+++==================
GUID:00d67812 tag for msg tracking for Spyder devops corresp.

From: Carlos Cordoba ***@***.***>
Sent: Thursday, December 29, 2022 3:30 PM
To: spyder-ide/spyder ***@***.***>
Cc: Steven A. Bender ***@***.***>; Mention ***@***.***>
Subject: EXTERNAL EMAIL : Re: [spyder-ide/spyder] Right click on editor tabs and selecting New window launches window then dies. (Issue #20275)


CAUTION: This email originated from outside of ICM. Do not click links or open attachments unless you recognize the sender and know the content is safe.

Hey @Sa-Bender17<https://github.com/Sa-Bender17>, thanks for reporting. This is a duplicate of issue #20101<https://github.com/spyder-ide/spyder/issues/20101> and it'll be fixed in our next version (5.4.1), to be released later today.

**Comment by ccordoba12** [comment]
> I love the tool, it’s quickly becoming my Python IDE of choice.

Thanks for your kind words @Sa-Bender17! Glad to know you enjoy using Spyder.
